source/nanomsg/wrap.d

```diff
@@ -239,6 +235,7 @@ struct NanoSocket {
 
         const sent = () @trusted { return nn_send(_nanoSock, data.ptr, data.length, flags(blocking)); }();
         if(blocking) {
+            enforceNanoMsgRet(sent, file, line);
```

Comment:
This is already covered by the exception being thrown below.

Reply:
this one checks for `>0` and checks errno (i.e. checking for nanomsg reporting failure), the next one checks for the exact length (i.e. nanomsg thinks it's fine but we're not happy with what it did)

Reply:
`>0` and same as the length is the same check except for when the length is 0, which is not something anybody should be doing anyway. And if they are, they should expect to send 0 bytes. Checking errno is a good idea, but then I'd change it to `enforceNanoMsgRet(sent == data.length, file, line)`.